src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuAutoDisapproveDocumentsServiceImpl.java

```diff
@@ -486,9 +387,37 @@ protected boolean checkIfDocumentEligibleForAutoDispproval(DocumentHeader docume
         
         return finalList;
     }
+    
+    /**
+     * This method finds the document for the given document header id
+     *
+     * @param documentHeaderId
+     * @return document The document in the workflow that matches the document header id.
+     */
+    protected Document findDocumentForAutoDisapproval(String documentHeaderId) {
+        Document document = null;
+
+        try {
+            document = getDocumentService().getByDocumentHeaderId(documentHeaderId);
+        } catch (WorkflowException ex) {
+            LOG.error("Exception encountered on finding the document: " + documentHeaderId, ex);
+        } catch (UnknownDocumentTypeException ex) {
+            // don't blow up just because a document type is not installed (but don't return it either)
+            LOG.error("Exception encountered on finding the document: " + documentHeaderId, ex);
+        }
+
+        return document;
+    }
 
     public void setRouteHeaderService(RouteHeaderService routeHeaderService) {
         this.routeHeaderService = routeHeaderService;
+    } 
+
+    protected PersonService getPersonService() {

```

Comment:
The superclass already has a non-final setter for its private personService variable. If that setter is being used properly by base KFS, perhaps it would be better to remove this getter, and to instead override the setter to call the superclass method and also set this subclass's personService variable. (Or, if this getter is still needed, then perhaps it would be better to perform the setter override above and to remove the lazy-load section from this getter.)